### PR TITLE
fix: support debug for sdk generator and add doc

### PR DESCRIPTION
**Pull request by pginoux-1A**
## Proposed change
* Support debug for sdk generator
* Add associated documentation

### CODE REVIEW

**Review by fpaul-1A — DISMISSED** [review]

File: packages/@ama-sdk/schematics/schematics/code-generator/open-api-cli-generator/open-api-cli.options.ts [hunk 1]
```diff
@@ -27,6 +27,14 @@ export type OpenApiCliOptions = CodegenTaskOptions & {
...
    */
   generatorCustomPath: string;
+  /**
+   * Comma separated string of options to give to the openapi-generator-cli
+   * Example: debugModels to log the full json structure used to generate models
+   * Example: debugOperations to log the full json structure used to generate operations
+   *
+   * @default ''
+   */
+  globalProperty: string;
```
Comment (kpanot) [inline packages/@ama-sdk/schematics/schematics/code-generator/open-api-cli-generator/open-api-cli.options.ts:37]: Maybe we should use plural to indicate the multi properties support: `globalProperties`.

File: packages/@ama-sdk/schematics/schematics/code-generator/open-api-cli-generator/open-api-cli.options.ts [hunk 2]
```diff
@@ -27,6 +27,14 @@ export type OpenApiCliOptions = CodegenTaskOptions & {
    * @default path to our own custom typescript generator
    */
   generatorCustomPath: string;
+  /**
+   * Comma separated string of options to give to the openapi-generator-cli
+   * Example: debugModels to log the full json structure used to generate models
```
Comment (kpanot) [inline packages/@ama-sdk/schematics/schematics/code-generator/open-api-cli-generator/open-api-cli.options.ts:32]: could be `@example`

File: packages/@ama-sdk/schematics/schematics/typescript/core/schema.ts [hunk 3]
```diff
@@ -12,4 +12,11 @@ export interface NgGenerateTypescriptSDKCoreSchematicsSchema extends JsonObject
 
   /** Path to the spec generation configuration */
   specConfigPath: string;
+
+  /**
+   * Comma separated string of options to give to the openapi-generator-cli
+   * Example: debugModels to log the full json structure used to generate models
+   * Example: debugOperations to log the full json structure used to generate operations
+   */
+  globalProperty: string;
```
Comment (kpanot) [inline packages/@ama-sdk/schematics/schematics/typescript/core/schema.ts:21]: Is it missing in `schema.json` on purpose or just forgot ?

Comment (pginoux-1A) [reply packages/@ama-sdk/schematics/schematics/typescript/core/schema.ts:21]: indeed :D

File: packages/@ama-sdk/schematics/schematics/code-generator/open-api-cli-generator/open-api-cli.options.ts (hunk 1 above)
Comment (pginoux-1A) [reply packages/@ama-sdk/schematics/schematics/code-generator/open-api-cli-generator/open-api-cli.options.ts:37]: That's the name of the variable taken by the generator, I agree that the plural would be better but I prefer keep the consistency here

**Review by kpanot — DISMISSED** [review]

File: packages/@ama-sdk/schematics/README.md [hunk 4]
```diff
@@ -75,3 +75,18 @@ yarn schematics @ama-sdk/schematics:java-client-core --spec-path ./swagger-spec.
 ```
 
 [Default swagger config](./schematics/java/client-core/swagger-codegen-java-client/config/swagger-codegen-config.json) will be used if `--swagger-config-path` is not provided.
+
+### Debug the typescript generator
+The open api generator extracts a JSON data model with a lot of information describing the specification provided, and use this data model to feed the templates and generate the code.
```
Comment (cpaulve-1A) [inline packages/@ama-sdk/schematics/README.md:80]:
```suggestion
The OpenApi generator extracts an enhance JSON data model from the specification yaml and use this data model to feed the templates to generate the code.
```

File: packages/@ama-sdk/schematics/README.md [hunk 5]
```diff
@@ -75,3 +75,18 @@ yarn schematics @ama-sdk/schematics:java-client-core --spec-path ./swagger-spec.
...
+If there is an issue with the files generated with the spec provided, the generator provides debugging features that log this data model.
+
+You can use global property options to pass one or both of the following options:
+* debugModel - log the full json structure used to generate models
```
Comment (cpaulve-1A) [inline packages/@ama-sdk/schematics/README.md:84]:
```suggestion
* debugModel - logs the full json structure used to generate models
```
@sdo-1A ?

File: packages/@ama-sdk/schematics/README.md (hunk 4 above)
Comment (sdo-1A) [reply packages/@ama-sdk/schematics/README.md:80]:
```suggestion
The OpenApi generator extracts an enhanced JSON data model from the specification YAML and uses this data model to feed the templates to generate the code.
```

File: packages/@ama-sdk/schematics/README.md [hunk 6]
```diff
@@ -75,3 +75,18 @@ yarn schematics @ama-sdk/schematics:java-client-core --spec-path ./swagger-spec.
...
+* debugOperations - log the full json structure used to generate operations
```
Comment (sdo-1A) [inline packages/@ama-sdk/schematics/README.md:85]:
```suggestion
* debugOperations - logs the full JSON structure used to generate operations
```

File: packages/@ama-sdk/schematics/README.md (hunk 5 above)
Comment (sdo-1A) [reply packages/@ama-sdk/schematics/README.md:84]: also is JSON better?

**Review by cpaulve-1A — DISMISSED** [review]

File: packages/@ama-sdk/schematics/schematics/typescript/core/index.ts [hunk 7]
```diff
@@ -44,6 +44,7 @@ export function ngGenerateTypescriptSDK(options: NgGenerateTypescriptSDKCoreSche
 
   const specPath = path.resolve(process.cwd(), options.specPath);
   const targetPath = options.directory || '/';
+  const globalProperty = options.globalProperty || '';
```
Comment (kpanot) [inline packages/@ama-sdk/schematics/schematics/typescript/core/index.ts:47]: `|| ''` not mandatory

**Review by kpanot — APPROVED** [review]

**Review by kpanot — APPROVED** [review]

**Review by sdo-1A — APPROVED** [review]

**Pull request merged**